csharp/ql/src/utils/modeleditor/ModelEditor.qll

```diff
@@ -129,3 +129,11 @@ private string qualifiedTypeName(string namespace, Type t) {
 private string qualifiedCallableName(string namespace, string type, Callable c) {
   exists(string name | hasQualifiedMethodName(c, namespace, type, name) | result = name)
 }
+
+/** A file that doesn't contain tests itself, but is only used in tests. */
+class TestSupportFile extends TestFile {
+  TestSupportFile() {
+    this.getAbsolutePath().matches(["%/test/%", "%/tests/%"]) and
+    not this.getAbsolutePath().matches("%/ql/test/%") // allows our test cases to work
+  }
```

Comment:
```suggestion
class TestRelatedFile extends File {
  TestRelatedFile() {
    this instanceof TestFile
    or
    this.getAbsolutePath().matches(["%/test/%", "%/tests/%"]) and
    not this.getAbsolutePath().matches("%/ql/test/%") // allows our test cases to work
  }
}
```

Reply:
This also requires that the usages of `TestFile` in the other places in this file is changed to `TestRelatedFile`.
My initial thought in the code review was merely to encapsulate the logic of whether one should consider a `File` a test like file in a single class to avoid duplicating (or have similar) logic in several predicates checking whether something was either a `TestFile` or `TestSupportFile`.

Reply:
Thanks, this makes sense. I've changed it to this.